src/tasks/LibraryBuilder/LibraryBuilder.cs

```diff
@@ -337,6 +337,10 @@ private string BuildAndroidLibrary(List<string> sources, List<string> libs, List
         buildOptions.CompilerArguments.Add(IsSharedLibrary ? $"-shared -o {libraryName}" : $"-o {libraryName}");
         buildOptions.IncludePaths.Add(MonoRuntimeHeaders);
         buildOptions.LinkerArguments.Add($"--soname={libraryName}");
+
+        // Google requires all the native libraries to be aligned to 16 bytes (for 16k memory page size)
+        // This is required only for 64-bit binaries, but there's not much harm if it is also done for 32-bit ones.
+        buildOptions.LinkerArguments.Add($"-z,max-page-size=16384");
```

Comment:
please also add this for NativeAOT to https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets#L242

you can use `Condition="'$(_linuxLibcFlavor)' == 'bionic'"` to check for android